distributed/tests/test_cancelled_state.py

```diff
@@ -231,53 +233,30 @@ async def wait_and_raise(*args, **kwargs):
         w.state.story(f1.key),
         [
             (f1.key, "executing", "released", "cancelled", {}),
-            (
-                f1.key,
-                "cancelled",
-                "error",
-                "error",
-                {f2.key: "executing", f1.key: "released"},
-            ),
-            (f1.key, "error", "released", "released", {f1.key: "forgotten"}),
+            (f1.key, "cancelled", "error", "released", {f1.key: "forgotten"}),
             (f1.key, "released", "forgotten", "forgotten", {}),
         ],
     )
 
 
-@gen_cluster(client=True, nthreads=[("", 1)])
-async def test_flight_cancelled_error(c, s, b):
-    """One worker with one thread. We provoke an flight->cancelled transition
-    and let the task err."""
-    lock = asyncio.Lock()
-    await lock.acquire()
-
-    class BrokenWorker(Worker):
-        block_get_data = True
+def test_flight_cancelled_error(ws):
```

Comment:
The previous test did not trip on #6877. I chose not to bother investigating why and just rewrote it.

Reply:
Because a test that does not test its declared use case is useless and misleading